test/sysmon/emitter_test.exs

```diff
@@ -0,0 +1,101 @@
+defmodule Instruments.EmitterTest do
+  use ExUnit.Case
+
+  alias Instruments.Sysmon.Emitter
+  alias Instruments.Sysmon.Reporter
+
+  defmodule TestEmitter do
+    @behaviour Instruments.Sysmon.Receiver
+
+    @impl true
+    def handle_busy_dist_port(pid, port) do
+      send(TestEmitterReceiver, {:busy_dist_port, pid, port})
+    end
+
+    @impl true
+    def handle_busy_port(pid, port) do
+      send(TestEmitterReceiver, {:busy_port, pid, port})
+    end
+
+    @impl true
+    def handle_long_gc(pid, info) do
+      send(TestEmitterReceiver, {:long_gc, pid, info})
+    end
+
+    @impl true
+    def handle_long_message_queue(pid, long) do
+      send(TestEmitterReceiver, {:long_message_queue, pid, long})
+    end
+
+    @impl true
+    def handle_long_schedule(pid, info) do
+      send(TestEmitterReceiver, {:long_schedule, pid, info})
+    end
+
+    @impl true
+    def handle_large_heap(pid, info) do
+      send(TestEmitterReceiver, {:large_heap, pid, info})
+    end
+  end
+
+  setup do
+    Process.register(self(), TestEmitterReceiver)
+    Application.put_env(:instruments, :sysmon_receiver, TestEmitter)
+
+    start_link_supervised!(Reporter)
+    pid = start_link_supervised!(Emitter)
```

Comment:
The minimum version on this library is low enough we can't even use `start_supervised!`. to preserve compatibility, we can do

```ex
{:ok, reporter_pid} = start_supervised(Reporter)
Process.link(reporter_pid)
{:ok, emitter_pid} = start_supervised(Emitter)
Process.link(emitter_pid)
```

(I don't believe the three-tuple value of `start_supervised` can be returned here https://stackoverflow.com/questions/13451588/where-does-info-in-return-value-ok-child-info-for-supervisorstart-child-com)